### PR TITLE
added protocol fee handling and verification to fork test

### PR DESCRIPTION
**Pull request by carlosdimatteo**

### CODE REVIEW

File: test/integration/arbitrum-mainnet/Loans.fork.t.sol [hunk 1]
```diff
@@ -141,48 +161,85 @@ contract LoansForkTest is LoansTestBase {
...
+            pair.rollsContract.previewRoll(rollOfferId, newPositionPrice);
+        uint providerBalanceBefore = pair.cashAsset.balanceOf(provider);
+        uint feeRecipientBalanceBefore = pair.cashAsset.balanceOf(feeRecipient);
         int minToUser = -1000e6; // Allow up to 1000 tokens to be paid by the user
         (uint newLoanId, uint newLoanAmount, int transferAmount) =
             rollLoan(pair, user, loanId, rollOfferId, minToUser);
 
+        // Verify fee taken and sent to recipient
+        assertEq(
+            pair.cashAsset.balanceOf(feeRecipient) - feeRecipientBalanceBefore, expectedResults.protocolFee
```
Comment (artdgn) [inline test/integration/arbitrum-mainnet/Loans.fork.t.sol:182]: would check that expected is nonzero here

File: test/integration/arbitrum-mainnet/Loans.fork.t.sol [hunk 2]
```diff
@@ -141,48 +161,85 @@ contract LoansForkTest is LoansTestBase {
...
+        IRolls.PreviewResults memory expectedResults =
+            pair.rollsContract.previewRoll(rollOfferId, pair.takerNFT.currentOraclePrice());
+
         (uint newLoanId, uint newLoanAmount, int transferAmount) =
-            rollLoan(pair, user, loanId, rollOfferId, minToUser);
+            rollLoan(pair, user, loanId, rollOfferId, -1000e6);
+
+        assertEq(
+            pair.cashAsset.balanceOf(feeRecipient) - recipientBalanceAfterOpenLoan,
+            expectedResults.protocolFee
```
Comment (artdgn) [inline test/integration/arbitrum-mainnet/Loans.fork.t.sol:216]: same, nonzero check make sense here

File: test/integration/arbitrum-mainnet/Loans.fork.t.sol [hunk 3]
```diff
@@ -141,48 +161,85 @@ contract LoansForkTest is LoansTestBase {
...
+        );
 
         // Advance time again
         vm.warp(block.timestamp + pair.durations[0]);
 
-        uint minUnderlyingOut = underlyingAmount * 95 / 100; // 5% slippage
+        uint minUnderlyingOut = underlyingAmount * (100 - slippage) / 100; // 5% slippage
```
Comment (artdgn) [inline test/integration/arbitrum-mainnet/Loans.fork.t.sol:222]: comment is incorrect

File: test/integration/arbitrum-mainnet/Loans.fork.t.sol [hunk 4]
```diff
@@ -141,48 +161,85 @@ contract LoansForkTest is LoansTestBase {
...
         uint underlyingOut = closeLoan(pair, user, newLoanId, minUnderlyingOut);
 
+        // Verify total protocol fees collected
+        assertEq(
+            pair.cashAsset.balanceOf(feeRecipient) - feeRecipientBalanceBefore,
+            initialFee + expectedResults.protocolFee
```
Comment (artdgn) [inline test/integration/arbitrum-mainnet/Loans.fork.t.sol:228]: nonzero

File: test/integration/arbitrum-mainnet/Loans.fork.t.sol [hunk 5]
```diff
@@ -141,48 +161,85 @@ contract LoansForkTest is LoansTestBase {
...
+        );
+
         assertGt(initialLoanAmount, 0);
-        assertGt(newLoanId, loanId);
         assertGe(int(newLoanAmount), int(initialLoanAmount) + transferAmount);
         assertGe(underlyingOut, minUnderlyingOut);
```
Comment (artdgn) [inline test/integration/arbitrum-mainnet/Loans.fork.t.sol:233]: this checks the return value and not the balance, so a bit less robust

File: test/integration/arbitrum-mainnet/Loans.fork.t.sol [hunk 6]
```diff
@@ -141,48 +161,85 @@ contract LoansForkTest is LoansTestBase {
...
     }
 
+    function getProviderProtocolFeeByUnderlying() internal view returns (uint protocolFee) {
+        // Calculate protocol fee based on post-swap provider locked amount
+        uint swapOut = underlyingAmount * pair.takerNFT.currentOraclePrice() / pair.oracle.baseUnitAmount();
+        uint initProviderLocked = swapOut * (callstrikeToUse - BIPS_BASE) / BIPS_BASE;
+        (protocolFee,) = pair.providerNFT.protocolFee(initProviderLocked, pair.durations[0]);
```
Comment (artdgn) [inline test/integration/arbitrum-mainnet/Loans.fork.t.sol:240]: would check here that fee is non zero

**Review by artdgn — APPROVED** [review]
a bunch of non-zero checks make sense, otherwise LGTM

File: test/integration/arbitrum-mainnet/Loans.fork.t.sol (hunk 3 above)
Comment (carlosdimatteo) [reply test/integration/arbitrum-mainnet/Loans.fork.t.sol:222]: i updated it and forgot about the comment 🤣

File: test/integration/arbitrum-mainnet/Loans.fork.t.sol (hunk 5 above)
Comment (carlosdimatteo) [reply test/integration/arbitrum-mainnet/Loans.fork.t.sol]: changed it

**Pull request merged**